src/app/profile-edit/profile-edit.page.ts

```diff
@@ -0,0 +1,139 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { LoaderService } from '../services/loader/loader.service';
+import { finalize, catchError } from 'rxjs/operators';
+import { of } from 'rxjs';
+import { ApiBaseService } from '../services/base-api/api-base.service';
+import { ToastService } from '../services/toast/toast.service';
+import urlConfig from 'src/app/config/url.config.json';
+import { FETCH_Profile_FORM } from '../core/constants/formConstant';
+import { MainFormComponent } from 'elevate-dynamic-form';
+@Component({
+  selector: 'app-profile-edit',
+  templateUrl: './profile-edit.page.html',
+  styleUrls: ['./profile-edit.page.scss'],
+})
+export class ProfileEditPage implements OnInit {
+  @ViewChild('formLib') formLib: MainFormComponent | undefined;
+  formJson: any = [];
+  urlProfilePath = urlConfig['profileListing']
+
+  constructor(
+    private apiBaseService: ApiBaseService,
+    private loader: LoaderService,
+    private toastService: ToastService
+  ) {}
+
+  ngOnInit() {
+    this.getFormJson();
+  }
+
+  async getFormJson() {
+    await this.loader.showLoading("Please wait while loading...");
+    this.apiBaseService.post(urlConfig['formListing'].listingUrl, FETCH_Profile_FORM)
+      .pipe(
+        finalize(() => this.loader.dismissLoading()),
+        catchError(err => {
+          this.toastService.presentToast(err?.error?.message);
+          return of(null);
+        })
+      )
+      .subscribe((res:any) => {
+        if (res?.status === 200) {
+          this.formJson = res?.result?.data;
+          this.formJson.forEach((control: any) => {
+            if (control.dynamicEntity) {
+              this.getOptionsData(control.name);
+            }
+          });
+        }
+      });
+  }
+
+  getOptionsData(entityType: string, entityId?: string) {
+    const control = this.formJson.find((control: any) => control.name === entityType);
+    if (!control) return;
+    const urlPath = control.dynamicEntity
```

Comment:
what if the dynamicEntity is true but its next in the hierarchy for example district? & if the dynamicEntity is false for some controls where its not required fetch the data from API it should not make an API then this condition would not be sufficient